handlers/user-password-reset-url.js

```diff
@@ -0,0 +1,133 @@
+/**
+ * user-password-reset-request
+ * our Request handler.
+ */
+
+const ABBootstrap = require("../AppBuilder/ABBootstrap");
+// {ABBootstrap}
+// responsible for initializing and returning an {ABFactory} that will work
+// with the current tenant for the incoming request.
+
+const { nanoid } = require("nanoid/async");
+
+module.exports = {
+   /**
+    * Key: the cote message key we respond to.
+    */
+   key: "user_manager.user-password-reset-url",
+
+   /**
+    * inputValidation
+    * define the expected inputs to this service handler:
+    * Format:
+    * "parameterName" : {
+    *    {joi.fn}   : {bool},  // performs: joi.{fn}();
+    *    {joi.fn}   : {
+    *       {joi.fn1} : true,   // performs: joi.{fn}().{fn1}();
+    *       {joi.fn2} : { options } // performs: joi.{fn}().{fn2}({options})
+    *    }
+    *    // examples:
+    *    "required" : {bool},
+    *    "optional" : {bool},
+    *
+    *    // custom:
+    *        "validation" : {fn} a function(value, {allValues hash}) that
+    *                       returns { error:{null || {new Error("Error Message")} }, value: {normalize(value)}}
+    * }
+    */
+   inputValidation: {
+      email: { string: { email: { allowUnicode: true } }, required: true },
+   },
+
+   /**
+    * fn
+    * our Request handler.
+    * @param {obj} req
+    *        the request object sent by the
+    *        api_sails/api/controllers/user_manager/user-password-reset-request.
+    * @param {fn} cb
+    *        a node style callback(err, results) to send data when job is finished
+    */
+   fn: function handler(req, cb) {
+      req.log("user_manager.user-password-reset-url:");
+
+      // get the AB for the current tenant
+      ABBootstrap.init(req)
+          .then(async (AB) => { // eslint-disable-line
+
+            req.log(req.param("email"));
+
+            // 1) get User
+            const cond = { email: req.param("email") };
+            const User = AB.objectUser();
+            const list = await req.retry(() =>
+               User.model().find({ where: cond, populate: false })
+            );
+
+            if (!list || !list[0]) {
+               // Q: is there any additional management in this case?
+               // eg: do we mark how many failed attempts and then block that browser?
+               req.log("User not found: ", JSON.stringify(cond));
+               cb(null, { status: "success" });
```

Comment:
should this be an error?  

I could not find the user that was requested.  This should at least respond with a "E_NOT_FOUND" sort of error.